Code/GraphMol/FileParsers/MolWriters.h

```diff
@@ -333,6 +343,71 @@ class RDKIT_FILEPARSERS_EXPORT PDBWriter : public MolWriter {
   unsigned int d_count;
   bool df_owner;
 };
+
+#ifdef RDK_BUILD_MAEPARSER_SUPPORT
+
+class RDKIT_FILEPARSERS_EXPORT MaeWriter : public MolWriter {
+  /**************************************************************************************
+   * A highly experimental Maestro file (or stream) writer. Many features are
+   * not supported yet, e.g. chirality and bond stereo, stereo groups, substance
+   * groups, isotopes or dummy atoms. Note that except for stereochemistry
+   * labels these aren't supported by the MaeMolSupplier either.
+   *
+   * Usage:
+   *  - create a MaeWriter with an output file name (or a ostream),
+   *     and a list of mol/atom/bond properties that need to be written out.
+   *     If no property names are specified, all properties will be exported.
+   *     Properties that are specified, but are not present will be ignored.
+   *  - then, a call is made to the write function for each molecule
+   *     that needs to be written out.
+   *
+   * Notes:
+   *  - kekulization is mandatory, as the Maestro format does not
+   *     have the concept of an aromatic bond.
+   *  - Ownership of the output stream is mandatory, since it needs
+   *     to be managed though a shared_ptr, as this is what maeparser
+   *     writer takes.
+   ***************************************************************************************/
+ public:
+  /*!
+    \param fileName       : filename to write to (stdout is *not* supported)
+   */
+  MaeWriter(const std::string &fileName);
```

Comment:
Can I annoyingly ask for 
```
static std::string getText(const ROMol &mol);
```

Reply:
I think you mean `MolToMaeBlock()`

Reply:
That's a separate function, btw. I'll implement that in a separate PR, this one is already big enough.

Reply:
Oh I guess I was looking at https://github.com/rdkit/rdkit/blob/27f711a6585e17c0c2882f6d7ea054a1f7a860eb/Code/GraphMol/FileParsers/MolWriters.h#L149 which would do the same as MolToMaeBlock? Or is there mae header stuff that MaeWriter::getText would return that MolToMaeBlock wouldn't?

Reply:
Yeah, all the work done there is this:
https://github.com/rdkit/rdkit/blob/4d4b92765197df1e60893858cc68f0eeb62adf0e/Code/GraphMol/FileParsers/SDWriter.cpp#L159-L164

... which is basically calling the free function `_MolToSDStream`, which in turn calls `MolToMolBlock`, and then adds some properties to the output:
https://github.com/rdkit/rdkit/blob/4d4b92765197df1e60893858cc68f0eeb62adf0e/Code/GraphMol/FileParsers/SDWriter.cpp#L113-L123